src/user.js

```diff
@@ -0,0 +1,169 @@
+import getAuthenticationDetails from './helpers/getAuthenticationDetails';
+import getCognitoUser from './helpers/getCognitoUser';
+import getUserPool from './helpers/getUserPool';
+import getCognitoUserAttributes from './helpers/getCognitoUserAttributes';
+
+const user = class User {
+  constructor(awsCognito) {
+    this.awsCognito = awsCognito;
+  }
+
+  auth(username, password, userPoolId, clientId) {
+    const authenticationDetails = getAuthenticationDetails(username, password,
+      this.awsCognito.AuthenticationDetails);
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.authenticateUser(authenticationDetails, {
+        onSuccess(result) {
+          resolve(result);
+        },
+        onFailure(err) {
+          reject(err);
+        },
+      });
+    });
+  }
+
+  create(username, password, userPoolId, clientId, attributes) {
+    const userPool = getUserPool(userPoolId, clientId, this.awsCognito.CognitoUserPool);
+    const cognitoUserAttributes = getCognitoUserAttributes(attributes,
+      this.awsCognito.CognitoUserAttribute);
+
+    return new Promise((resolve, reject) => {
+      userPool.signUp(username, password, cognitoUserAttributes, null, (err, result) => {
+        if (err) {
+          return reject(err);
+        }
+
+        return resolve(result);
+      });
+    });
+  }
+
+  confirmRegistration(username, confirmationCode, userPoolId, clientId) {
+    const cognitoUser = getCognitoUser(username, userPoolId, clientId,
+      this.awsCognito.CognitoUserPool, this.awsCognito.CognitoUser);
+
+    return new Promise((resolve, reject) => {
+      cognitoUser.confirmRegistration(confirmationCode, true, (err, result) => {
+        if (err) {
```

Comment:
The following code pattern appears quite a few places in your code.  I would consider extracting that into a reusable function that could allow you in future to implement logging in a centralized place. 

```
(err, result) => {
   if (err) {
     return reject(err);
   }
   return resolve(result);
 }
```
Here is how that function could look. For the sake of clarity im not using arrow functions, but I would suggest you do
```
function genericResponse(resolve, reject) {
     return function(err, result) {
           if(err) {
                return reject(err);
            }
           return resolve(result);
      }
}
```
You implementation code would then look something like this
```
cognitoUser.confirmRegistration(confirmationCode, true, genericResponse(resolve, reject));
```